### PR TITLE
Fix subject parsing

### PR DESCRIPTION
**Pull request by kwapik**

### CODE REVIEW

File: src/sources/azure_blob/queue.rs [hunk 1]
```diff
@@ -160,57 +161,69 @@ async fn proccess_event_grid_message(
...
-    }
-    let blob = parts[6];
-    info!(
-        "New blob created in container '{}': '{}'",
-        &container, &blob
-    );
+    match parse_subject(body.subject) {
+        Some((container, blob)) => {
+            if container != container_client.container_name() {
+                trace!("Container name does not match. Skipping.");
```
Comment (ldrozdz93) [inline src/sources/azure_blob/queue.rs:167]: This should be either an error or a warning. It's not a sunny day path.

File: src/sources/azure_blob/queue.rs [hunk 2]
```diff
@@ -160,57 +161,69 @@ async fn proccess_event_grid_message(
...
+                            .pop_receipt_client(message)
+                            .delete()
+                            .await
+                            .expect("Failed removing messages from queue");
+                    })
+                }),
+            })
+        }
+        None => {
+            trace!("Failed parsing subject. Skipping.");
```
Comment (ldrozdz93) [inline src/sources/azure_blob/queue.rs]: This should be either an error or a warning. It's not a sunny day path.

**Review by ldrozdz93 — APPROVED** [review]

File: src/sources/azure_blob/queue.rs (hunk 1 above)
Comment (kwapik) [reply src/sources/azure_blob/queue.rs:167]: Well, in our setup it is acceptable. Event subscriptions are per storage account, not container. So we can get events for other containers as well. IMO having it as trace is ok.

File: src/sources/azure_blob/queue.rs (hunk 2 above)
Comment (kwapik) [reply src/sources/azure_blob/queue.rs]: Will change to warn.

**Pull request merged**